### PR TITLE
added speech synthesizers

### PR DESCRIPTION
**Pull request by ZimLim**
* created component for text to speech
* used SpeechSyntesisUtterance as it is widely available on all web browser
* added button that play the speech audio when clicked
* speeches are synthesized in order of button clicked

### CODE REVIEW

File: src/components/TextToSpeech.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import PlayArrowIcon from '@mui/icons-material/PlayArrow';
+import { SxProps, SystemProps } from '@mui/system';
+
+
+
+export const TextToSpeech: FC<TypographyProps> = (props) => {
+
+    const [audioAvailable, setAudioAvailable] = useState(false);
+    const [iconSize, setIconSize] = useState(16);
+    const [message, setMessage] = useState("");
```
Comment (gregfrasco) [inline src/components/TextToSpeech.tsx:12]: this is not needed you can just pass props down

File: src/components/TextToSpeech.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+        <Box
+            display="flex"
+            flexDirection={"row"}
+        >
+            <Typography {...props}/>
+            {audioAvailable &&
+                <Button
+                    onClick={handleClick}
+                    endIcon={<PlayArrowIcon style={{fontSize : iconSize}}/>}
```
Comment (gregfrasco) [inline src/components/TextToSpeech.tsx:49]:
```suggestion
                    endIcon={<PlayArrowIcon style={{fontSize : props.iconSize || 16}}/>}
```

File: src/components/TextToSpeech.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { FC, ReactElement, useEffect, useState, ReactNode } from "react";
+import { Box, Button, Typography, TypographyProps } from "@mui/material";
+import PlayArrowIcon from '@mui/icons-material/PlayArrow';
+import { SxProps, SystemProps } from '@mui/system';
+
+
```
Comment (gregfrasco) [inline src/components/TextToSpeech.tsx:6]: if you need your own props you can declare them here:
```suggestion
export interface TextToSpeechProps extends TypographyProps {
iconSize?: number;
} 
```

File: src/components/TextToSpeech.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+export const TextToSpeech: FC<TypographyProps> = (props) => {
```
Comment (gregfrasco) [inline src/components/TextToSpeech.tsx:8]: Then switch out your props
```suggestion
export const TextToSpeech: FC<TextToSpeechProps> = (props) => {
```

File: src/components/TextToSpeech.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        if(props.children){
+            setMessage(String(props.children))
+        }
+
+    }, [])
+
+    const handleClick = () => {
+        var msg = new SpeechSynthesisUtterance();
+        msg.text = message;
```
Comment (gregfrasco) [inline src/components/TextToSpeech.tsx:33]: Then this would be props.children

File: src/components/TextToSpeech.tsx (hunk 3 above)
Comment (ZimLim) [reply src/components/TextToSpeech.tsx]: decided to only use props passed down by typography

File: src/components/TextToSpeech.tsx (hunk 4 above)
Comment (ZimLim) [reply src/components/TextToSpeech.tsx:5]: decided to only use props passed down by typography

**Pull request closed without merging**